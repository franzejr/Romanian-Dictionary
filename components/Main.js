import React, { Component } from 'react';
import api  from '../API';
import DefinitionsList from './DefinitionsList';

import {
  View,
  Text,
  ScrollView,
  StyleSheet,
  TextInput,
  Button,
  ActivityIndicator
} from 'react-native';

var styles = StyleSheet.create({
  mainContainer: {
    flex: 1,
    padding: 20,
    flexDirection: 'column',
    justifyContent: 'center',
    backgroundColor: '#5f9ea0'
  },
  title: {
    marginTop: 20,
    marginBottom: 40,
    fontSize: 25,
    textAlign: 'center',
    color: '#fff'
  },
  searchInput: {
    height: 50,
    padding: 4,
    marginRight: 5,
    fontSize: 24,
    borderWidth: 3,
    borderColor: 'white',
    color: 'white'
  },
});

class Main extends Component {
  constructor(props){
    super(props);
    this.state = {
      word: '',
      definitions: [],
      isLoading: false,
      error: false
    }
  }
  handleChange(event){
    this.setState({
      word: event.nativeEvent.text
    })
  }
  handleSubmit(){
    this.setState({
      isLoading: true,
    });
    api.getWord(this.state.word)
      .then((jsonRes) => this.handleResponse(jsonRes))
      .catch((err) => {
        this.setState({
          isLoading: false,
          error: `There was an error: ${err}`
        })
      })
  }
  handleResponse(res){
    this.setState({
      definitions: res.definitions,
      isLoading: false,
      error: false
    });
  }
  render() {
    var definitions = this.state.definitions;
    var showErr = (
      this.state.error ? <Text> {this.state.error} </Text> : <View></View>
    );
    return(
      <View style={styles.mainContainer}>
        <Text style={styles.title}> ROMANIAN DICTIONARY </Text>
        <TextInput
          style={styles.searchInput}
          value={this.state.word}
          placeholder="Search here"
          onChange={this.handleChange.bind(this)} />
        <Button
          onPress={this.handleSubmit.bind(this)}
          color="#ffff"
          accessibilityLabel="Search for a word in Romanian"
          title="Search" />
        <ActivityIndicator
          animating={this.state.isLoading}
          color="#111"
          size="large"></ActivityIndicator>
        {showErr}

        <ScrollView
          ref={(scrollView) => { _scrollView = scrollView; }}
          automaticallyAdjustContentInsets={false}
          onScroll={() => { console.log('onScroll!'); }}
          scrollEventThrottle={200}
          style={styles.scrollView}>
          <DefinitionsList definitions={definitions} />
        </ScrollView>
      </View>
      )
  }
};

export default Main;
