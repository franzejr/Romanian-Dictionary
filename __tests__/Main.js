import 'react-native';
import React from 'react';
import Main from '../components/Main';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Main />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
