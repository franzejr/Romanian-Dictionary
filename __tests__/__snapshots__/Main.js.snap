exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#5f9ea0",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "padding": 20,
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#fff",
        "fontSize": 25,
        "marginBottom": 40,
        "marginTop": 20,
        "textAlign": "center",
      }
    }>
     ROMANIAN DICTIONARY 
  </Text>
  <TextInput
    onChange={[Function]}
    placeholder="Search here"
    style={
      Object {
        "borderColor": "white",
        "borderWidth": 3,
        "color": "white",
        "fontSize": 24,
        "height": 50,
        "marginRight": 5,
        "padding": 4,
      }
    }
    value="" />
  <TouchableOpacity
    accessibilityComponentType="button"
    accessibilityLabel="Search for a word in Romanian"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    activeOpacity={0.2}
    disabled={undefined}
    onPress={[Function]}>
    <View
      style={
        Array [
          Object {},
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#ffff",
            },
          ]
        }>
        Search
      </Text>
    </View>
  </TouchableOpacity>
  <ActivityIndicator
    animating={false}
    color="#111"
    hidesWhenStopped={true}
    size="large" />
  <View />
  <ScrollView
    automaticallyAdjustContentInsets={false}
    onScroll={[Function]}
    scrollEventThrottle={200}
    style={undefined}>
    <View
      style={undefined} />
  </ScrollView>
</View>
`;
